sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionAsyncClient.java

```diff
@@ -0,0 +1,268 @@
+    private static final DefaultJsonSerializer DEFAULT_SERIALIZER = new DefaultJsonSerializer();
+
+    private final IngestionUsingDataCollectionRulesAsyncClient service;
+
+    LogsIngestionAsyncClient(IngestionUsingDataCollectionRulesAsyncClient service) {
+        this.service = service;
+    }
+
+    /**
+     * Uploads logs to Azure Monitor with specified data collection rule id and stream name. The input logs may be
+     * too large to be sent as a single request to the Azure Monitor service. In such cases, this method will split
+     * the input logs into multiple smaller requests before sending to the service.
+     * @param dataCollectionRuleId the data collection rule id that is configured to collect and transform the logs.
+     * @param streamName the stream name configured in data collection rule that matches defines the structure of the
+     * logs sent in this request.
+     * @param logs the collection of logs to be uploaded.
+     * @return the result of the logs upload request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<UploadLogsResult> upload(String dataCollectionRuleId, String streamName, List<Object> logs) {
+        return upload(dataCollectionRuleId, streamName, logs, new UploadLogsOptions());
+    }
+
+    /**
+     * Uploads logs to Azure Monitor with specified data collection rule id and stream name. The input logs may be
+     * too large to be sent as a single request to the Azure Monitor service. In such cases, this method will split
+     * the input logs into multiple smaller requests before sending to the service.
+     * @param dataCollectionRuleId the data collection rule id that is configured to collect and transform the logs.
+     * @param streamName the stream name configured in data collection rule that matches defines the structure of the
+     * logs sent in this request.
+     * @param logs the collection of logs to be uploaded.
+     * @param options the options to configure the upload request.
+     * @return the result of the logs upload request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<UploadLogsResult> upload(String dataCollectionRuleId, String streamName,
+                                         List<Object> logs, UploadLogsOptions options) {
+        return withContext(context -> upload(dataCollectionRuleId, streamName, logs, options, context));
+    }
+
+    Mono<UploadLogsResult> upload(String dataCollectionRuleId, String streamName,
+                                  List<Object> logs, UploadLogsOptions options,
+                                  Context context) {
+        return Mono.defer(() -> splitAndUpload(dataCollectionRuleId, streamName, logs, options, context));
+    }
+
+    private Mono<UploadLogsResult> splitAndUpload(String dataCollectionRuleId, String streamName, List<Object> logs, UploadLogsOptions options, Context context) {
+        try {
```

Comment:
by GA, let' figure out how the tracing experience would look like - we'd need one outer span per `upload`